src/delete_selective.lua

```diff
@@ -68,14 +72,27 @@ function plugindef()
         Deletes nominated items from the selected region. 
         Individual menu items are created to independently delete items of type:  
         All Expressions / Dynamics / Expressions (Not Dynamics) / Expressions (Measure-Attached) /  
-        Articulations / Hairpins / Slurs / Custom Lines / Glissandos / Smart Shapes (Beat Aligned) /  
-        All Smart Shapes / Midi Note Data / Midi Continuous Data
+        Articulations / Lyrics / Hairpins / Slurs / Custom Lines / Glissandos /  
+        Smart Shapes (Beat Aligned) / All Smart Shapes / Midi Note Data / Midi Continuous Data  
     ]]
     return "Delete All Expressions", "Delete All Expressions", "Delete all expressions from the selected region"
 end
 
 delete_type = delete_type or "expression_all"
 
+function expression_is_dynamic(exp)
+    if not exp:IsShape() and exp.Visible and exp.StaffGroupID == 0 then
+        local cd = finale.FCCategoryDef()
+        local text_def = exp:CreateTextExpressionDef()
+        if text_def and cd:Load(text_def.CategoryID) then
+            if text_def.CategoryID == finale.DEFAULTCATID_DYNAMICS or string.find(cd:CreateName().LuaString, "Dynamic") then
```

Comment:
Use `cd.Type` instead of `text_def.CategoryID`. This calculation happens several times throughout the repository. I'm thinking it should be a library function in the expression library. A Lua-version-independent approach would be:

```lua
if text_def.CategoryID == finale.DEFAULTCATID_DYNAMICS then
   return true
end
if cd.Type == finale.DEFAULTCATID_DYNAMICS then
   return true
end
if string.find(cd:CreateName().LuaString, "Dynamic") then
   return true
end
return false
```

However, this script cannot run a version of FinaleLua that lacks `cd.Type`, so the first check is not needed here.

Reply:
> Use `cd.Type` instead of `text_def.CategoryID`.

Many thanks Robert. Correction follows.

Reply:
In the modified version, `cd` is not loaded. I think the definition has a `Create...` method to do this.

Reply:
Oops. Thanks for the eagle eyes.